calyx/src/backend/verilog/gen.rs

```diff
@@ -326,9 +327,10 @@ fn wire_id_from_port<'a>(port: &Port) -> D<'a> {
         Port::Comp { component, port } => {
             D::text(format!("{}_{}", component.to_string(), port.to_string()))
         }
-        Port::Hole { .. } => unreachable!(
-            "This should have been caught in the validation checking"
-        ),
+        Port::Hole { group, name } => unreachable!(format!(
+            "Structure has a group hole: {}[{}]",
+            group.id, name.id
+        )),
```

Comment:
Note for @sgpthomas: It is not caught by the validator for some reason. Remind me to write a test case.

Reply:
Specifically, I made the changes to the inliner that @sgpthomas needs to review.

Reply:
where are the changes to the inliner? are they in this branch?

Reply:
Yeah, they should be. Specifically the inlined wouldn’t delete holes if there were no reads from them.